lib/Raven/Client.php

```diff
@@ -233,35 +238,49 @@ class Client
      */
     protected $_shutdown_function_has_been_set;
 
-    public function __construct($options_or_dsn = null, $options = [])
+    /**
+     * @var Configuration The client configuration
+     */
+    protected $config;
+
+    /**
+     * Constructor.
+     *
+     * @param Configuration $config The client configuration
+     */
+    public function __construct(Configuration $config)
```

Comment:
> the client depends on the configuration to work

Yes. And you can create it in constructor
```php
function __construct(array $options = []) {
    $configuration = new Configuration($options);
    // @todo ...
}
```

instead of demanding do it from our users